internal/cli/cmd/sysdump.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
+
+	"github.com/cilium/cilium-cli/sysdump"
+)
+
+var (
+	sysdumpOptions = sysdump.SysdumpOptions{
+		LargeSysdumpAbortTimeout: sysdump.DefaultLargeSysdumpAbortTimeout,
+		LargeSysdumpThreshold:    sysdump.DefaultLargeSysdumpThreshold,
+		Writer:                   sysdump.DefaultWriter,
+	}
+)
+
+func newCmdSysdump() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sysdump",
+		Short: "Collects information required to troubleshoot issues with Cilium and Hubble",
+		Long:  ``,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Silence klog to avoid displaying "throttling" messages - those are expected.
+			klog.SetOutput(ioutil.Discard)
+			// Collect the sysdump.
+			if err := sysdump.NewSysdump(k8sClient, sysdumpOptions).Collect(); err != nil {
+				return fmt.Errorf("failed to collect sysdump: %v", err)
+			}
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVar(&contextName, "context", "", "Kubernetes configuration context")
+	cmd.Flags().StringVar(&sysdumpOptions.CiliumLabelSelector, "cilium-label-selector", sysdump.DefaultCiliumLabelSelector, "The labels used to target Cilium pods")
```

Comment:
it's an interesting concept (using an intermediary struct to collect all the options together), but I find this whole section really messy and hard to read primarily to the length of each line.

A good layout for cobra options that I have found is pretty much using the usual golang line breaks with a few minor tweaks:

```
cmd.Flags().<TYPE>Var<P?>(&location.Var, // can be in-line since for reader it often doesn't matter
    "my-custom-var", defaults.DefaultValue, // name clearly visible with default
    "This is a custom var to do xyz", // description also clear on it's own line
)
.. other flags
```

This allows for easy section scan to see all the possible values (I think)

Reply:
This makes total sense :+1: I'm pushing the change shortly.

Reply:
Done!